common/common-client/src/main/java/org/ow2/proactive/authentication/iam/IAMRestClient.java

```diff
@@ -0,0 +1,123 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.authentication.iam;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.ow2.proactive.http.CommonHttpClientBuilder;
+
+
+public class IAMRestClient {
+
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(IAMRestClient.class.getName());
+
+    private IAMRestClient() {
+
+    }
+
+    public static synchronized String getSSOTicket(String url, String username, String password, String ticketHeader) {
+
+        String ticket = null;
+
+        try (CloseableHttpClient httpClient = new CommonHttpClientBuilder().allowAnyCertificate(true).build()) {
+
+            HttpPost httpPost = new HttpPost(url);
+
+            List<NameValuePair> params = new ArrayList<>();
+            params.add(new BasicNameValuePair("username", username));
+            params.add(new BasicNameValuePair("password", password));
+
+            httpPost.setEntity(new UrlEncodedFormEntity(params));
+
+            HttpResponse httpResponse = httpClient.execute(httpPost);
+
+            if ((httpResponse.getStatusLine().getStatusCode() != HttpStatus.SC_CREATED) ||
+                (null == httpResponse.getEntity()))
+                throw new IAMException("Authentication failed");
+
+            Header[] headers = httpResponse.getHeaders(ticketHeader);
+
+            if (headers[0] != null) {
+
+                Pattern pattern = Pattern.compile("(TGT-\\d+-\\S+)");
+                Matcher matcher = pattern.matcher(headers[0].toString());
+
+                if (matcher.find()) {
+                    ticket = matcher.group(1);
+                } else
+                    throw new IAMException("TGT not found in IAM response");
+            }
+
+            LOG.debug("SSO ticket successfully generated for user " + username + "\n Token: " + ticket);
+
+        } catch (Exception e) {
+            throw new IAMException(e.getMessage());
+        }
+
+        return ticket;
+    }
+
+    public static synchronized String getServiceToken(String url, String service) {
```

Comment:
I use the keywords 'static' and 'synchronized' because I choosed to use a single connection to IAM at once for memory consumption (sequential authentication). But yes I changed my mind and made modifications to enable parallel authentication.